src/main/java/org/mule/service/http/impl/service/server/grizzly/HttpParser.java

```diff
@@ -45,6 +50,14 @@ public static String extractPath(String uri) {
     return path;
   }
 
+  public static String decodePath(String path) throws MuleRuntimeException {
```

Comment:
Since this is new here we can take the opportunity to fail with a proper `Exception`. Perhaps `DecodeException`?